### PR TITLE
BREAKING(yaml): rename `ParseOptions.json` to `ParseOptions.allowDuplicateKeys`

### PR DESCRIPTION
**Pull request by iuioiua**
### What's changed

[`ParseOptions.json`](https://jsr.io/@std/yaml@0.224.3/doc/parse/~/ParseOptions.json) has been renamed `ParseOptions.allowDuplicateKeys`. This change only affects those that previously set `ParseOptions.json` to `true`. No changes in behavior have been made.

### Motivation

This name more clearly describes how the option affects the behavior of [`parse()`](https://jsr.io/@std/yaml/doc/~/parse).

### Migration guide

To migrate, use the `allowDuplicateKeys` option instead of the `json` option for `parse()`.

```diff
import { parse } from "@std/yaml/parse";

const yamlString = `name: John Doe
age: 30
name: Jane Doe`;

- parse(yamlString, { json: true });
+ parse(yamlString, { allowDuplicateKeys: true });
```

### Related

Towards #5195

### CODE REVIEW

**Comment by kt3k** [comment]
I'm in favor of this change. The option name `json` doesn't describe what it changes.

I slightly feel `allowDuplicatedKeys` is a bit more natural. What do you think?

**Comment by iuioiua** [comment]
> I slightly feel `allowDuplicatedKeys` is a bit more natural. What do you think?

I don't think it's any more natural than what it is now.

**Comment by iuioiua** [comment]
I did a quick search - "duplicate keys" seems more correct.

File: yaml/parse.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export interface ParseOptions {
   /** Name of the schema to use.*/
   schema?: "core" | "default" | "failsafe" | "json" | "extended";
   /** compatibility with JSON.parse behaviour. */
```
Comment (kt3k) [inline yaml/parse.ts:16]: Let's also change the description

Comment (iuioiua) [reply yaml/parse.ts:16]: I'll be reworking docs for all of `@std/yaml` once the breaking changes are made.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**